modules/nf-core/peddy/main.nf

```diff
@@ -0,0 +1,56 @@
+process PEDDY {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/peddy:0.4.8--pyh5e36f6f_0' :
+        'biocontainers/peddy:0.4.8--pyh5e36f6f_0' }"
+
+    input:
+    tuple val(meta), path(vcf), path(vcf_tbi)
+    path ped
+
+    output:
+    tuple val(meta), path("*.html")     , emit: html
+    tuple val(meta), path("*.csv")      , emit: csv
+    tuple val(meta), path("*.peddy.ped"), emit: ped
+    tuple val(meta), path("*.png")      , optional: true, emit: png
+    path "versions.yml"                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    peddy \\
+        $args \\
+        --prefix $prefix \\
```

Comment:
I have removed the `--plot` flag here as it crashed when running on a single sample

Would be better it this could be handled through configuration I think, but did not seem possible at the moment

Reply:
I see that the nf-core tests are failing due to module mismatch. So this would be something to change in the nf-core module itself I suspect?

Reply:
I wonder, doesn't the same issue appear for single sample runs in the raredisease pipeline? Or maybe single samples are not run through Peddy over there 🤔 

I.e.

```
$ cat my.ped
#family_id      sample  father  mother  sex     phenotype
hg002_case      hg002   0       0       M       affected
```

```
$ singularity exec <path>/depot.galaxyproject.org-singularity-peddy-0.4.8--pyh5e36f6f_0.img peddy --plot my.vcf.gz my.ped
```

```
2024-12-11 08:30:31 rs-fe1 peddy.cli[108576] INFO Running Peddy version 0.4.8
2024-12-11 08:30:31 rs-fe1 peddy.cli[108576] INFO ped_check
2024-12-11 08:30:33 rs-fe1 peddy.peddy[108576] INFO plotting
2024-12-11 08:30:33 rs-fe1 matplotlib.legend[108576] WARNING No handles with labels found to put in legend.
2024-12-11 08:30:34 rs-fe1 peddy.cli[108576] INFO ran in 2.7 seconds
2024-12-11 08:30:34 rs-fe1 peddy.cli[108576] INFO het_check
2024-12-11 08:30:35 rs-fe1 peddy.pca[108576] INFO loaded and subsetted thousand-genomes genotypes (shape: (2504, 23)) in 0.5 seconds
2024-12-11 08:30:37 rs-fe1 peddy.pca[108576] INFO ran randomized PCA on thousand-genomes samples at 23 sites in 1.2 seconds
2024-12-11 08:30:37 rs-fe1 peddy.pca[108576] INFO Projected thousand-genomes genotypes and sample genotypes and predicted ancestry via SVM in 0.0 seconds 
/usr/local/lib/python3.9/site-packages/peddy/peddy.py:888: RuntimeWarning: Mean of empty slice.
  bot = depth[depth > bot].mean() - 5 * depth[depth > bot].std()
/usr/local/lib/python3.9/site-packages/numpy/core/_methods.py:261: RuntimeWarning: Degrees of freedom <= 0 for slice
  ret = _var(a, axis=axis, dtype=dtype, out=out, ddof=ddof,
Traceback (most recent call last):
...
  File "/usr/local/lib/python3.9/site-packages/matplotlib/path.py", line 634, in get_extents
    xys = np.concatenate(xys)
  File "<__array_function__ internals>", line 5, in concatenate
ValueError: need at least one array to concatenate
```

While after removing `--plot`

```
$ singularity exec <path>/depot.galaxyproject.org-singularity-peddy-0.4.8--pyh5e36f6f_0.img peddy my.vcf.gz my.ped
```

It runs fine

Reply:
I did a patch for this for now, removing `--plot`